lib/timeout.rb

```diff
@@ -132,6 +132,12 @@ def self.ensure_timeout_thread_created
         unless @timeout_thread and @timeout_thread.alive?
           @timeout_thread = create_timeout_thread
         end
+
+        # shut down timeout thread at exit
+        Kernel.at_exit do
+          @timeout_thread.kill
```

Comment:
How about closing the queue and checking that? It seems nicer than killing the thread.

Reply:
I switched from an external kill to an external queue close + internal kill and it still works well.